server/src/lib/model/db.ts

```diff
@@ -224,6 +224,7 @@ export default class DB {
             WHERE skipped.sentence_id = sentences.id AND
               skipped.client_id = ?
           )
+          AND (clips_count = 0 OR has_valid_clip = 0)
```

Comment:
This is to ensure the column gets backfilled in the right order - when this feature gets released, `has_valid_clip` will be zero for everything, including sentences with thousands of clips, and we still want clips with lower clip counts to be prioritized in sentence serving. Since most languages have a lot of sentences with a clips_count of zero, that should shortcircuit the condition, but for languages where all sentences have been recorded at least once (e.g. rw) then it'll fall back to the regular `has_valid_clip`. 

(It's possible there's a mistake in my logic somewhere.)

Reply:
Okay, so I'm working through this slooowly and still can't understand what's being shortcircuited, but I think I'm probably just missing something cool about optimization. And tbh, I can live with that!

> and we still want clips with lower clip counts to be prioritized in sentence serving

that's how I interpreted the next line, `ORDER BY clips_count ASC`, but I'm still confused about how the `OR` helps. BUT, it's a fuzzy brain rainy day, so unless you're feeling exceedingly patient and have nothing better to do (and… I've seen your calendar 🤯), I think we should land it as-is.

Reply:
AH sorry, this is totally a case of me jumping too far ahead - I would like to get rid of the `ORDER BY clips_count` condition at some point, since sorting is always such a heavy database burden and with a fully backfilled single sentence limit we wouldn't need this sort condition at all because it wouldn't matter how many clips currently exist for a sentence if none of them are valid. If we didn't have the sort condition, then being able to jump directly to sentences with a clips_count of 0 matters, even though right now it doesn't and is technically redundant...does that make sense? Again, I could be totally off-base here!

Reply:
AH sorry, this is totally a case of me jumping too far ahead - I would like to get rid of the ORDER BY clips_count condition at some point, since sorting is always such a heavy database burden and with a fully backfilled single sentence limit we wouldn't need this sort condition at all because it wouldn't matter how many clips currently exist for a sentence if none of them are valid. If we didn't have the sort condition, then being able to jump directly to sentences with a clips_count of 0 matters, even though right now it doesn't and is technically redundant...does that make sense? Again, I could be totally off-base here!

Reply:
I just thought of a WAY clearer way of saying what I meant initially; sorry for the back-and-forth. Since `clips_count = 0` is a subset of `has_valid_clip = 0`, I think:

```sql
AND (clips_count = 0 OR has_valid_clip = 0)
```

will always yield the same results as:

```sql
AND has_valid_clip = 0
```

---

Unless there's a case I'm missing where `has_valid_clip` can be `1` without any clips recorded, but that seems like it would be a bug.